core/scripts.py

```diff
@@ -3563,6 +3563,66 @@ def _update_list_with_new_entries(self, custom_list: CustomList):
         custom_list.auto_update_status = CustomList.UPDATED
 
 
+class DeleteInvisibleLanesScript(LibraryInputScript):
+    """Delete lanes that are flagged as invisible including any visible sub-lanes."""
+
+    def process_library(self, library):
+
+        try:
+            continue_update = True
+            while continue_update:
```

Comment:
I'm not understanding this `while` loop. In what cases do we have to loop here?

Reply:
I verified that cascading is happening.  I reworked the PR to remove cruft.